tel2puml/find_unique_graphs/otel_ingestion/otel_data_source.py

```diff
@@ -11,6 +14,38 @@
 class OTELDataSource(ABC):
     """Abstract class for returning a OTelEvent object from a data source."""
 
+    def __init__(self) -> None:
+        """Constructor method."""
+        self.yaml_config = self.get_yaml_config()
+        self.valid_data_sources = ["json"]
+        self.data_source = self.set_data_source()
+
+    def get_yaml_config(self) -> Any:
```

Comment:
I think we just want to load in the data source options rather than a direct link to the config file to load in - we will already have had to read the config file to ascertain what data source we want so we can just use the options under that data source input as a typed dict (also will allow us to check the config is correct). The config file will be user input I would have thought rather than a stock thing for the repo (although we can have a default). So I think I would remove this function